packages/vsx-registry/src/node/vsx-extension-resolver.ts

```diff
@@ -74,16 +75,24 @@ export class VSXExtensionResolver implements PluginDeployerResolver {
                 return;
             }
         }
-        const downloadPath = (await this.environment.getExtensionsDirUri()).path.fsPath();
-        await fs.ensureDir(downloadPath);
-        const extensionPath = path.resolve(downloadPath, path.basename(downloadUrl));
-        console.log(`[${resolvedId}]: trying to download from "${downloadUrl}"...`, 'to path', downloadPath);
-        if (!await this.download(downloadUrl, extensionPath)) {
+        const downloadDir = await this.getTempDir();
+        await fs.ensureDir(downloadDir);
```

Comment:
Do we clean this up somewhere?

Reply:
Now we do ;)
I adjusted the `PluginVsCodeFileHandler` to cleanup `pluginEntry().originalPath()` if it is a temp dir.